src/thorlabs_cube/driver/tcube/tsc.py

```diff
@@ -0,0 +1,407 @@
+
+        # Send the message with the packed payload for MOT_SET_SOL_CYCLEPARAMS (0x04C3)
+        await self.send(Message(MGMSG.MOT_SET_SOL_CYCLEPARAMS, data=payload))
+
+    async def get_solenoid_cycle_parameters(self):
+        """Get the current solenoid cycle parameters.
+
+        :return: A tuple containing (on_time, off_time, num_cycles).
+        """
+
+        # Pack the Channel ID for the request
+        payload = st.pack("<H", 1)
+
+        # Send the request to get the current solenoid cycle parameters
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_SOL_CYCLEPARAMS, [MGMSG.MOT_GET_SOL_CYCLEPARAMS], data=payload
+        )
+
+        on_time, off_time, num_cycles = st.unpack("<LLL", get_msg.data[2:])
+
+        return on_time, off_time, num_cycles
+
+    async def set_sol_interlock_mode(self, mode):
+        """Set the solenoid interlock mode.
+
+        :param mode: Interlock mode, where:
+                    - 0x01 = SOLENOID_ENABLED (hardware interlock required)
+                    - 0x02 = SOLENOID_DISABLED (hardware interlock not required)
+        """
+
+        # Send the message with Channel ID in param1 and Interlock Mode in param2
+        await self.send(Message(MGMSG.MOT_SET_SOL_INTERLOCKMODE, param2=mode))
+
+    async def get_sol_interlock_mode(self):
+        """Get the current solenoid interlock mode.
+
+        :return: The interlock mode, where:
+                - 0x01 = SOLENOID_ENABLED (hardware interlock required)
+                - 0x02 = SOLENOID_DISABLED (hardware interlock not required)
+        """
+
+        # Send the request to get the current interlock mode
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_SOL_INTERLOCKMODE, [MGMSG.MOT_GET_SOL_INTERLOCKMODE], param1=1
+        )
+
+        interlock_mode = get_msg.param2
+
+        return interlock_mode
+
+    async def set_sol_state(self, state):
+        """Set the solenoid state (ON or OFF).
+
+        :param state: The solenoid state, where:
+                    - 0x01 = SOLENOID_ON (solenoid is active).
+                    - 0x02 = SOLENOID_OFF (solenoid is deactivated).
+        """
+
+        # Send the message with Channel ID in param1 and State in param2
+        await self.send(Message(MGMSG.MOT_SET_SOL_STATE, param2=state))
+
+    async def get_sol_state(self):
+        """Get the current solenoid state.
+
+        :return: The solenoid state, where:
+                - 0x01 = SOLENOID_ON (solenoid is active).
+                - 0x02 = SOLENOID_OFF (solenoid is deactivated).
+        """
+
+        # Send the request to get the current solenoid state
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_SOL_STATE, [MGMSG.MOT_GET_SOL_STATE], param1=1
+        )
+
+        # Return the solenoid state from param2
+        return get_msg.param2
+
+
+class TscSim:
+    def module_identify(self):
+        pass
+
+    def get_bay_used(self):
+        return False
```

Comment:
The comments indicate that the original get_bay_used function when the KSC is connected to a Thorlabs Hub is `Identify which bay is being used by the controller on Thorlabs Hub`. Since in sim, there is no physical hub being used, I thought it would just be easier to return False. Would it be better to return a different datatype such as a string indicating that get_bay_used returns a string `sim` so it is more explicit for the user?

Reply:
This method should return a value that has the same datatype as the `get_bay_used()` method in the `Tsc` driver class.